src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -250,16 +268,13 @@ private MatrixFactorizationTrainer(IHostEnvironment env, Arguments args) : base(
         /// <param name="matrixColumnIndexColumnName">The name of the column hosting the matrix's column IDs.</param>
         /// <param name="matrixRowIndexColumnName">The name of the column hosting the matrix's row IDs.</param>
         /// <param name="labelColumn">The name of the label column.</param>
-        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
-        public MatrixFactorizationTrainer(IHostEnvironment env,
+        internal MatrixFactorizationTrainer(IHostEnvironment env,
```

Comment:
>internal [](start = 8, length = 8)

Can you and @sfilipi take habit of marking this internal constructors as BestFriend?